### PR TITLE
Added additional final states and ACR/ACS/DocDB to azure project.

### PR DESCRIPTION
**Pull request by alvadb**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/ResourceManagement/DocumentDB/Microsoft.Azure.Management.DocumentDB.Fluent/DatabaseAccountImpl.cs [hunk 1]
```diff
@@ -89,7 +90,15 @@ private Models.DatabaseAccountCreateUpdateParametersInner CreateUpdateParameters
 
         private bool IsAFinalProvisioningState(string state)
```
Comment (hovsepm) [inline src/ResourceManagement/DocumentDB/Microsoft.Azure.Management.DocumentDB.Fluent/DatabaseAccountImpl.cs]: >A [](start = 23, length = 1)

We should any article. A or The.

Comment (martinsawicki) [inline src/ResourceManagement/DocumentDB/Microsoft.Azure.Management.DocumentDB.Fluent/DatabaseAccountImpl.cs]: the more standard sounding naming I think would be `IsProvisioningStateFinal(...)`

File: global.json [hunk 2]
```diff
@@ -1,33 +1,37 @@
...
+    "Samples/ResourceManagement/Network",
+    "Samples/ResourceManagement/RedisCache",
+    "Samples/ResourceManagement/ResourceManager",
+    "Samples/ResourceManagement/ServiceBus",
+    "Samples/ResourceManagement/Sql",
+    "Samples/ResourceManagement/Storage",
+    "Samples/ResourceManagement/TrafficManager",
+    "src/ResourceManagement/ContainerRegistry",
+    "src/ResourceManagement/DocumentDB"
+  ]
```
Comment (hovsepm) [inline global.json]: please delete these two lines they are already added above.

File: Samples/ResourceManagement/DocumentDB/CreateDocumentDBWithEventualConsistency/Program.cs [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+                // Create a new collection.
+                myCollection = documentClient.CreateDocumentCollectionAsync(
+                        "dbs/" + DATABASE_ID, myCollection, requestOptions)
+                        .GetAwaiter().GetResult();
+            }
+            catch (Exception ex)
+            {
+                throw ex;
+            }
```
Comment (hovsepm) [inline Samples/ResourceManagement/DocumentDB/CreateDocumentDBWithEventualConsistency/Program.cs]: no need to catch to just re-throw.

File: Samples/ResourceManagement/DocumentDB/ManageHADocumentDB/Program.cs [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+                RequestOptions requestOptions = new RequestOptions();
+                requestOptions.OfferThroughput = 4000;
+
+                // Create a new collection.
+                myCollection = documentClient.CreateDocumentCollectionAsync(
+                        "dbs/" + DATABASE_ID, myCollection, requestOptions)
+                        .GetAwaiter().GetResult();
+            } catch (Exception ex) {
+                throw ex;
+            }
```
Comment (hovsepm) [inline Samples/ResourceManagement/DocumentDB/ManageHADocumentDB/Program.cs]: Same here. No need to catch to just rethrow.

**Comment by hovsepm** [comment]
> using Xunit;

please add license header. 

---
Refers to: src/ResourceManagement/Azure.Fluent/Fluent.Tests/Compute/ContainerServicesTest.cs:1 in 89373d8. [](commit_id = 89373d8ffdedef7f5ab52398f258bd11f51f6a69, deletion_comment = False)

**Comment by hovsepm** [comment]
> using Xunit;

License header missing

---
Refers to: src/ResourceManagement/Azure.Fluent/Fluent.Tests/ContainerRegistry/ContainerRegistriesTest.cs:1 in 89373d8. [](commit_id = 89373d8ffdedef7f5ab52398f258bd11f51f6a69, deletion_comment = False)

File: src/ResourceManagement/Azure.Fluent/Fluent.Tests/DocumentDB/DocumentDBTest.cs [hunk 5]
```diff
@@ -0,0 +1,68 @@
+﻿using Fluent.Tests.Common;
+using Microsoft.Azure.Management.DocumentDB.Fluent;
```
Comment (hovsepm) [inline src/ResourceManagement/Azure.Fluent/Fluent.Tests/DocumentDB/DocumentDBTest.cs]: License header missing.

**Pull request merged**